content/docs/reference/design-docs/rolling-stock-collections.en.md

```diff
@@ -0,0 +1,64 @@
+---
+title: "Rolling stock collections"
+linkTitle: "Rolling stock collections"
+weight: 90
+---
+
+## Context and requirements
+
+As of the writting of this proposal, rolling stocks:
+- are identified using a globally unique name
+- cannot be made private
+
+With the upcoming authorization system, this setup bring a couple of new issues:
+- timetables can have a lot of different rolling stocks, which would need individual permission checks if the current setup were to be preserved
+- users probably do not want to have to individually give access to each rolling stock in a large collection
+- looking up rolling stocks becomes a per-user job, as not all users are authorized to know a given rolling stock exists
+
+These issues are deemed too damaging to implement rolling stock authorization as is.
+
+## Proposal
+
+A concept of rolling stock collection would be introduced:
+- each rolling stock must belong to exactly one collection
+- rolling stocks do not have individual permissions; permissions are managed at the collection level
+- when creating a scenario, rolling stock collections need to be added
+
+Rolling stock collections have:
+- a name
+- a description
+
+
+```sql
+create table rolling_stock_collection(
+  id  bigserial generated always as identity primary key,
+  name text not null,
+  created_by  bigint references authn_user on delete set null,
+  created_at  timestamp not null default CURRENT_TIMESTAMP,
+);
+
+alter table rolling_stock
+add column collection bigserial references rolling_stock_collection on delete cascade not null;
+
+-- speed up cascade
+create index on rolling_stock_collection(created_by);
+create index on rolling_stock(collection);
+```
+
```

Comment:
I guess we also would like to add a `rolling_stock_collection_id` to the `timetable` table, isn't it?